### PR TITLE
ARROW-5362: [C++] Fix compression test memory usage

### PR DESCRIPTION
**Pull request by emkornfield**

### CODE REVIEW

File: cpp/src/arrow/util/compression-test.cc [hunk 1]
```diff
@@ -69,37 +69,39 @@ void CheckCodecRoundtrip(Compression::type ctype, const std::vector<uint8_t>& da
   int64_t actual_size;
   ASSERT_OK(c1->Compress(data.size(), data.data(), max_compressed_len, compressed.data(),
                          &actual_size));
-  compressed.resize(actual_size);
```
Comment (pitrou) [inline cpp/src/arrow/util/compression-test.cc]: I'm not sure I understand. Why do you remove this?

**Review by pitrou — COMMENTED** [review]
It would be nice to know more about the motivation for these changes.

File: cpp/src/arrow/util/compression-test.cc (hunk 1 above)
Comment (emkornfield) [reply cpp/src/arrow/util/compression-test.cc]: I use returned sizes below, rather the explicit vector size.   

Snappy appears to use more than the final compressed size of the buffer when compressing which triggers:

https://github.com/google/sanitizers/wiki/AddressSanitizerContainerOverflow

Comment (pitrou) [reply cpp/src/arrow/util/compression-test.cc]: You mean when decompressing?

Comment (emkornfield) [reply cpp/src/arrow/util/compression-test.cc]: No compressing.  The problem is actually below on the second compress round.  This seemed like a cleaner fix then resizing up again before the second compression (but I suppose it prevents sanitizer verification of decompression not consuming more bytes then it should)

Comment (pitrou) [reply cpp/src/arrow/util/compression-test.cc]: I think resizing again before the second compression would make the intent clearer.

Comment (emkornfield) [reply cpp/src/arrow/util/compression-test.cc]: Sounds good, I'll do that

Comment (emkornfield) [reply cpp/src/arrow/util/compression-test.cc]: @pitrou made the changes.

**Review by pitrou — APPROVED** [review]
+1, thanks

**Pull request closed without merging**